core/src/main/java/io/methvin/watchservice/MacOSXListeningWatchService.java

```diff
@@ -145,6 +147,8 @@ public AbstractWatchKey register(
             flags);
 
     final CFRunLoopThread thread = new CFRunLoopThread(streamRef, file.toFile());
+    callback.setRunLoopThread(thread);
```

Comment:
It seems like the only reason the callback needs a reference to the `CFRunLoopThread` is because it needs to call the `close` method on this class. Perhaps we could pass another callback here, like this:

```java
final CFRunLoopThread thread =
  new CFRunLoopThread(
    streamRef,
    file.toFile(),
    () -> close(thread, callback, file)
  );
```

Reply:
Nevermind, I wasn't reading the code right here 🤦 . But I think there might be a better way.